lib/jeckle/rest_actions.rb

```diff
@@ -5,19 +5,105 @@ def self.included(base)
     end
 
     module Collection
+      # @public
+      #
+      # The root name that Jeckle will parse the <b>response</b>. Default is <b>false</b>.
+      #
+      # @example
+      #
+      #   module Dribble
+      #     class Shot
+      #       include Jeckle::Resource
+      #       root collection: true, member: true
+      #     end
+      #   end
+      #
+      #   Shot.collection_root_name # => Will parse the root node as 'shots'
+      #   Shot.member_root_name # => Will parse the root node as 'shot'
+      #
+      #  Sometimes that are APIs you need to fetch /projects, BUT the root node is extremely different from the resource name.
+      #
+      #   module OtherApi
+      #     class Project
+      #       include Jeckle::Resource
+      #       api :my_api
+      #       root collection: 'awesome-projects', member: 'awesome-project'
+      #     end
+      #   end
+      #
+      def root(options={})
+        @collection_root_name = find_root_name(options[:collection], :pluralize)
+        @member_root_name     = find_root_name(options[:member], :singularize)
+      end
+
+      # @public
+      #
+      # Member action that requests for the resource using the resource name
+      #
+      #  @example
+      #
+      #     Post.find(1) # => posts/1
+      #
       def find(id)
-        endpoint = "#{resource_name}/#{id}"
-        attributes = run_request(endpoint).response.body
+        endpoint   = "#{resource_name}/#{id}"
+        response   = run_request(endpoint).response.body
+        attributes = parse_response(response, member_root_name)
```

Comment:
Don't bother with this. After `action` macro all "default" methods (find, search, etc) will be built with it, this implementation will have a short living